pkg/controllers/securitypolicy_controller.go

```diff
@@ -27,13 +28,35 @@ import (
 	"github.com/vmware-tanzu/nsx-operator/pkg/util"
 )
 
-var (
-	log = logf.Log.WithName("controller").WithName("securitypolicy")
-)
-
 const (
 	WCP_SYSTEM_RESOURCE = "vmware-system-shared-t1"
 	METRIC_RES_TYPE     = "securitypolicy"
+	MAX_WORKER_COUNT    = 5
+)
+
+var (
+	log           = logf.Log.WithName("controller").WithName("securitypolicy")
+	resultNormal  = ctrl.Result{}
+	resultRequeue = ctrl.Result{Requeue: true}
+
+	updateFail = func(r *SecurityPolicyReconciler, c *context.Context, o *v1alpha1.SecurityPolicy, e *error) {
+		r.setSecurityPolicyReadyStatusFalse(c, o, e)
+		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateFailTotal, METRIC_RES_TYPE)
+	}
+
+	deleteFail = func(r *SecurityPolicyReconciler, c *context.Context, o *v1alpha1.SecurityPolicy, e *error) {
+		r.setSecurityPolicyReadyStatusFalse(c, o, e)
+		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteFailTotal, METRIC_RES_TYPE)
+	}
+
+	updateSuccess = func(r *SecurityPolicyReconciler, c *context.Context, o *v1alpha1.SecurityPolicy) {
+		r.setSecurityPolicyReadyStatusTrue(c, o)
+		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateSuccessTotal, METRIC_RES_TYPE)
+	}
+
+	deleteSuccess = func(r *SecurityPolicyReconciler, c *context.Context, o *v1alpha1.SecurityPolicy) {
+		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteSuccessTotal, METRIC_RES_TYPE)
+	}
```

Comment:
these four var (including deleteSuccess) seems just a normal function, how about take them out of var block?

Reply:
ok